bloom-housing/bloom
# fix: prevent public users from logging into partners

# Pull Request Template

## Issue Overview

This PR addresses #3226


## Description

There is currently an issue in production where a user can create an account in the public site then go to the partner site and successfully login. They are not able to see any data, but it is not best practice to allow this.

This PR blocks the actual login of the public users by seeing if the login origin url is the partner url and comparing it to the partners url setup in the environment variable

## How Can This Be Tested/Reviewed?

1. Reseed the backend
2. Create an account on the public site
3. Logout and go to the partner site
4. Try to log in on the partner site
5. You should get an error message

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
